config/checkmk-agent/checkmk.inc

```diff
@@ -39,24 +40,61 @@ function checkmk_install() {
 	*/
 	$checkmk_bin = "/usr/local/bin/check_mk_agent";
 	$checkmk_url = 'http://git.mathias-kettner.de/git/?p=check_mk.git;a=blob_plain;f=agents/check_mk_agent.freebsd;hb=e13899bde8bdafe13780427811c8153c59be807f';
-	mwexec("fetch -o {$checkmk_bin} \"{$checkmk_url}\"");
+	mwexec("/usr/bin/fetch -o {$checkmk_bin} \"{$checkmk_url}\"");
 	chmod($checkmk_bin, 0755);
 
-	sync_package_checkmk();
+	/* Detect possible junk left over after previous bad package versions */
+	checkmk_decrapify();
+}
+
+function checkmk_deinstall() {
+	/* Remove entry from /etc/services file */
+	checkmk_cleanup_etc_services_file();
+	/* Remove check_mk_agent script fetched via checkmk_install() */
+	unlink_if_exists("/usr/local/bin/check_mk_agent");
+	/* Detect possible junk left over after previous bad package versions */
+	checkmk_decrapify();
+}
+
+function checkmk_decrapify() {
+	$i = 0;
+	if (exec("/usr/bin/wc -l /etc/hosts.allow | /usr/bin/awk '{ print $1 }'") > 5) {
+		log_error("[check_mk-agent] Possibly redundant lines found in /etc/hosts.allow.");
+		$i++;
+	}
+	if (exec("/usr/bin/wc -l /etc/inetd.conf | /usr/bin/awk '{ print $1 }'") > 1) {
+		log_error("[check_mk-agent] Possibly redundant lines found in /etc/inetd.conf.");
+		$i++;
+	}
+	if (file_exists("/etc/rc.conf.local")) {
+		log_error("[check_mk-agent] /etc/rc.conf.local file found; this file does not exist normally on pfSense.");
+		$i++;
+	}
+	if ($i > 0) {
+		log_error("[check_mk-agent] Inconsistent configuration files; possibly caused by previous check_mk package versions.");
+		log_error("[check_mk-agent] Please, compare those against default distribution files at https://github.com/pfsense/pfsense and fix as required manually.");
+	}
```

Comment:
Probably, even though I suspect there are exactly 2 users of this package - one who filed the Redmine ticket and another who PMed me on the forum asking how to use files from this PR. (And I'm not even sure whether it's not the same person :eyes: )